lib/Runtime/Library/JavascriptObject.cpp

```diff
@@ -2174,7 +2178,7 @@ BOOL JavascriptObject::DefineOwnPropertyHelper(RecyclableObject* obj, PropertyId
         // TODO: implement DefineOwnProperty for other object built-in exotic types.
         else
         {
-            returnValue = JavascriptOperators::DefineOwnPropertyDescriptor(obj, propId, descriptor, throwOnError, scriptContext);
+            returnValue = JavascriptOperators::DefineOwnPropertyDescriptor(obj, propId, descriptor, throwOnError, scriptContext, Js::PropertyOperation_StrictMode);
```

Comment:
Why should this always be strict mode?

Reply:
As far as I understood, calls to `Object.defineXXX` should always throw on error (be strict) opposed to using the assignment operator (`property = value`).

The error is actually only thrown, if `throwOnError == true`.
https://github.com/chakra-core/ChakraCore/blob/fcdbd7b83d12b6ed07a1173192e9e46586ce8bfc/lib/Runtime/Library/JavascriptProxy.cpp#L1762

Reply:
indeed, i'm pretty sure all builtins always are effectively ran in strict mode

Reply:
I guess it seems strange as (in my understanding) `PropertyOperation_StrictMode`in CC code base always meant “strict mode” has been turned on explicitly; That’s reflected by CC error messages like
https://github.com/chakra-core/ChakraCore/blob/320a7a819e9c7ce7d5837bbe8f600d75069207f7/lib/Parser/rterrors.h#L155
The error messages throw by me do not explicitly state “strict mode” (According to the behaviour of `node`).

Reply:
The strict mode check in the Proxy Trap seems like it may be wrong - this call should throw if it's strict mode or not.

Trying to think if there's any case where throwOnError would be true and strict mode would be false and it should NOT throw - what could actually get there?

Reply:
You could use the assignment operator in non-strict mode; that should not throw.

Reply:
But would the throwOnError parameter be true at that point?

Reply:
Let me have a look...

Reply:
It seems to be like `throwOnError` is `true` in both scenarios so we need a seperate way to check for strict mode.